src/Windows/R/Editor/Impl/Settings/REditorSettings.cs

```diff
@@ -24,15 +26,28 @@ public sealed class REditorSettings: EditorSettings, IWritableREditorSettings, I
 
         private RFormatOptions _formatOptions = new RFormatOptions();
 
-        public REditorSettings(ICoreShell coreShell): base(coreShell, RContentTypeDefinition.ContentType) { }
-        public REditorSettings(IEditorSettingsStorage storage) : base(storage) { }
+        public REditorSettings(ICoreShell coreShell) : base(coreShell, RContentTypeDefinition.ContentType) {
+            CreateLintOptions();
+        }
+
+        public REditorSettings(IEditorSettingsStorage storage) : base(storage) {
+            CreateLintOptions();
+        }
 
         public override void ResetSettings() {
             _formatOptions = new RFormatOptions();
-            LintOptions = new LintOptions();
+            CreateLintOptions();
             base.ResetSettings();
         }
 
+        private void CreateLintOptions() {
+            LintOptions = new LintOptions();
+            LintOptions.PropertyChanged += _OnLintOptionsPropertyChanged;
```

Comment:
So theoretically storages can be reduced to two: lang preferences for the common editor stuff and roaming. Basically override DialogPage storage and map it to roaming. Import/Export settings won't work but since it is a roaming storage settings will travel. But requires logging on to VS account since otherwise roaming won't work...